compiler/rustc_lint/src/context.rs

```diff
@@ -720,6 +720,8 @@ impl<'tcx> LateContext<'tcx> {
     /// Anonymous scopes such as `extern` imports are matched with `kw::Invalid`;
     /// inherent `impl` blocks are matched with the name of the type.
     ///
+    /// If possible, consider using a `rustc_diagnostic_item` instead.
```

Comment:
"possible" is a bit vague. This should only be done when it really is about diagnostics.